### ISSUE TITLE
[BUG]Unable to delete a host on Harvester

### ISSUE THREAD

**Issue by DoNotSayYes** [harvester/harvester#463]
**Describe the bug**
Unable to delete a host on Harvester

**To Reproduce**
1. Launch harvester
2. Add up hosts to 3
3. Delete one of the hosts

**Result:**
The host was not removed accordingly and displayed unavailable. Seeing the node not ready in the cluster.
![image](https://user-images.githubusercontent.com/57614736/106722036-6dbd6180-6640-11eb-8ff0-0285d33d526f.png)
![image](https://user-images.githubusercontent.com/57614736/106722100-83328b80-6640-11eb-902e-23297ace355b.png)

**Environment:**
 - Harvester ISO version: master-head. (3dd7670)
 - Installation Mode: ISO/app-mode
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**
Related k3s issue:
https://github.com/k3s-io/k3s/issues/2669

**Comment by noahgildersleeve** [comment]
This is still an issue on `master-3b6b8a6d-head` . I tried to manually remove the node from rancher and it still failed. I attached the supportBundle for it. For me the issue was when the node was in unavailable status due to being turned off. However when I deleted another host it went through fine. I was told that it might be related to this issue. https://github.com/k3s-io/k3s/issues/2669

[harvester-supportbundle_046bfe97-0e0e-45e2-b9e2-6ed767e0aa1d_2021-06-01T22-36-14Z.zip](https://github.com/harvester/harvester/files/6580933/harvester-supportbundle_046bfe97-0e0e-45e2-b9e2-6ed767e0aa1d_2021-06-01T22-36-14Z.zip)
